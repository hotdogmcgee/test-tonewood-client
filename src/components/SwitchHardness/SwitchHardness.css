
#SwitchHardness__Section {
    width: 80vw;
    align-self: center;
}
.switch-container {
    display: flex;
    justify-content: center;
}

.hardness-switches {
    justify-content: space-between;
}

.switch-wood-type {
    margin-right: 5px;
}

  /* Switch element */
  .react-switch-checkbox {
    height: 0;
    width: 0;
    visibility: hidden;
  }
  
  .react-switch-label {
    display: flex;
    align-items: center;
    justify-content: space-between;
    cursor: pointer;
    width: 50px;
    height: 25px;
    background: grey;
    border-radius: 100px;
    position: relative;
    transition: background-color .2s;
  }
  
  .react-switch-label .react-switch-button {
    content: '';
    position: absolute;
    top: 2px;
    left: 2px;
    width: 20px;
    height: 20px;
    border-radius: 45px;
    transition: 0.2s;
    background: #fff;
    box-shadow: 0 0 2px 0 rgba(10, 10, 10, 0.29);
  }
  
  .react-switch-checkbox:checked + .react-switch-label .react-switch-button {
    left: calc(100% - 2px);
    transform: translateX(-100%);
  }
  
  .react-switch-label:active .react-switch-button {
    width: 30px;
  }

  @media screen and (min-width: 700px) {

    #SwitchHardness__Section {
      width: 50vw;
    }
    .hardness-switches {
      justify-content: space-around;
    }

    .switch-wood-type {
      margin-right: 10px;
      font-size: 25px;
    }

    #SwitchHardness__Section .react-switch-label {
      margin-bottom: 0;
      top: 3px;
      align-self: center;
    }
  }