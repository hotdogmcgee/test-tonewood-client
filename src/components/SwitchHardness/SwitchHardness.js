import React from "react";
import { Switch } from "../Utils/Utils";
import "./SwitchHardness.css";

export default class SwitchHardness extends React.Component {
  static defaultProps = {
    softwoodOn: "",
    hardwoodOn: "",
    handleSoftwoodChange: () => {},
    handleHardwoodChange: () => {}
  };

  render() {
    return (
      <div className="hardness-switches">
        <Switch
          isOn={this.props.softwoodOn}
          handleChange={this.props.handleSoftwoodChange}
          switchId={"react-switch-soft"}
          type={"Softwood"}
        />
        <Switch
          isOn={this.props.hardwoodOn}
          handleChange={this.props.handleHardwoodChange}
          switchId={"react-switch-hard"}
          type={"Hardwood"}
        />
      </div>
    );
  }
}
