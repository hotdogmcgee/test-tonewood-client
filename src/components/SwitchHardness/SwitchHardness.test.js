import React from "react";
import SwitchHardness from "./SwitchHardness";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("SwitchHardness component", () => {
  it("renders a SwitchHardness by default", () => {
    const wrapper = shallow(<SwitchHardness />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

//implement testing for clicking a Switch
});
