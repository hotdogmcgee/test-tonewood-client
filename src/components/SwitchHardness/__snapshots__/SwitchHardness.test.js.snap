// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchHardness component renders a SwitchHardness by default 1`] = `
<div
  className="hardness-switches"
>
  <Switch
    handleChange={[Function]}
    isOn=""
    switchId="react-switch-soft"
    type="Softwood"
  />
  <Switch
    handleChange={[Function]}
    isOn=""
    switchId="react-switch-hard"
    type="Hardwood"
  />
</div>
`;
